projects/driver_controls/src/gpio_expander.c

```diff
@@ -0,0 +1,168 @@
+#include "gpio_expander.h"
+
+#include <stdbool.h>
+
+#include "i2c.h"
+#include "gpio_it.h"
+#include "mcp23008.h"
+
+typedef struct GPIOExpanderInterrupt {
+  GPIOExpanderCallback callback;
+  void *context;
+} GPIOExpanderInterrupt;
+
+static GPIOAddress s_address;
+static I2CPort s_i2c_port;
+
+static GPIOExpanderInterrupt s_interrupts[NUM_GPIO_EXPANDER_PIN];
+
+// Default register values (Datasheet Table 1-3)
+static uint8_t s_register[NUM_MCP23008_REGISTERS] = {
```

Comment:
I mean, you seem to read them every time anyways. Most registers never change - is it really that important to store them?